### PR TITLE
Added test cases in greedy_algorithms\kruskals_minimum_spanning_tree.cpp

### PR DESCRIPTION
**Pull request by Monalisa0311**
#### Description of Change
Added the test cases for  greedy_algorithms\kruskals_minimum_spanning_tree.cpp issue #2509 

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Added the test cases in greedy_algorithms\kruskals_minimum_spanning_tree.cpp to validate edge cases.

### CODE REVIEW

**Comment by Monalisa0311** [comment]
If any changes are needed, let me know.

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp [hunk 1]
```diff
@@ -50,20 +56,167 @@ void findMinimumEdge(const int &infinity,
...
+                                             5,
+                                             0,
+                                             INFINITY_INT,
+                                             INFINITY_INT,
+                                             INFINITY_INT,
+                                             INFINITY_INT,
+                                             7,
+                                             INFINITY_INT,
+                                             0};
+    greedy_algorithms::findMinimumEdge(INFINITY_INT, graph1);
```
Comment (realstealthninja) [inline greedy_algorithms/kruskals_minimum_spanning_tree.cpp:111]: could you format this better?

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp [hunk 2]
```diff
@@ -50,20 +56,167 @@ void findMinimumEdge(const int &infinity,
...
-        1,        3,        0,        INFINITY, 1,        INFINITY,
-        4,        8,        INFINITY, 0,        5,        7,
-        INFINITY, 3,        1,        5,        0,        INFINITY,
-        INFINITY, INFINITY, INFINITY, 7,        INFINITY, 0};
-
-    greedy_algorithms::findMinimumEdge(INFINITY, graph);
+    test();  // run Self-test implementation
     return 0;
-}
+}
```
Comment (realstealthninja) [inline greedy_algorithms/kruskals_minimum_spanning_tree.cpp:222]:
```suggestion
}

```

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp [hunk 3]
```diff
@@ -50,20 +56,167 @@ void findMinimumEdge(const int &infinity,
...
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+static void test() {
+    // Test case with interger values
+    std::cout << "\nTest Case 1 :\n";
+    std::array<std::array<int, 6>, 6> graph1{0,
+                                             4,
```
Comment (realstealthninja) [inline greedy_algorithms/kruskals_minimum_spanning_tree.cpp:76]: add an uint32_t test case,

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Comment by Monalisa0311** [comment]
I changed the formatting. can you please review it?

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp (hunk 1 above)
Comment (Monalisa0311) [reply greedy_algorithms/kruskals_minimum_spanning_tree.cpp]: I change the formatting

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp (hunk 3 above)
Comment (Monalisa0311) [reply greedy_algorithms/kruskals_minimum_spanning_tree.cpp]: added the unit 32 test case

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp (hunk 2 above)
Comment (Monalisa0311) [reply greedy_algorithms/kruskals_minimum_spanning_tree.cpp]: added a line at the end

**Comment by Monalisa0311** [comment]
I have made all the required changes. Can you please review the changes?

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp [hunk 4]
```diff
@@ -50,20 +57,132 @@ void findMinimumEdge(const int &infinity,
...
+ * @returns void
+ */
+static void test() {
+    // Test case with interger values
+    std::cout << "\nTest Case 1 :\n";
+    std::array<std::array<int, 6>, 6> graph1{
+             0,            4,            1,             4,        INFINITY_INT,   INFINITY_INT,
+             4,            0,            3,             8,             3,         INFINITY_INT,
+             1,            3,            0,        INFINITY_INT,       1,         INFINITY_INT,
+             4,            8,       INFINITY_INT,        0,            5,              7,
```
Comment (realstealthninja) [inline greedy_algorithms/kruskals_minimum_spanning_tree.cpp:82]: could you follow the same formatting for the rest of the test cases as well?

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp (hunk 4 above)
Comment (Monalisa0311) [reply greedy_algorithms/kruskals_minimum_spanning_tree.cpp:82]: I changed it. Can you review it?

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp [hunk 5]
```diff
@@ -50,20 +57,132 @@ void findMinimumEdge(const int &infinity,
...
+/**
+ * define a large constant value for int
+ * define a large constant value for float
+ * define a large constant value for double
+ * define infinity for uint32_t
+ */
+constexpr int INFINITY_INT = std::numeric_limits<int>::max();
+constexpr float INFINITY_FLOAT = std::numeric_limits<float>::max();
+constexpr double INFINITY_DOUBLE = std::numeric_limits<double>::max();
+constexpr uint32_t INFINITY_UINT32 = UINT32_MAX;
```
Comment (realstealthninja) [inline greedy_algorithms/kruskals_minimum_spanning_tree.cpp:69]: Move these into main

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: greedy_algorithms/kruskals_minimum_spanning_tree.cpp (hunk 5 above)
Comment (Monalisa0311) [reply greedy_algorithms/kruskals_minimum_spanning_tree.cpp]: made the changes

**Review by realstealthninja — APPROVED** [review]

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Review by realstealthninja — APPROVED** [review]

**Comment by realstealthninja** [comment]
Thank you very much for your contributions :heart:
